core/sdk-core/src/main/java/software/amazon/awssdk/core/checksums/Algorithm.java

```diff
@@ -16,14 +16,15 @@
 package software.amazon.awssdk.core.checksums;
 
 import java.util.Map;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
 import software.amazon.awssdk.annotations.SdkPublicApi;
 import software.amazon.awssdk.utils.StringUtils;
 import software.amazon.awssdk.utils.internal.EnumUtils;
 
 /**
  * Enum that indicates all the checksums supported by Flexible checksums in a Service Request/Response Header.
  */
-@SdkPublicApi
+@SdkProtectedApi
```

Comment:
Moving things from public to protected seems unsafe, since people may have already read it and taken a dependency on it. In the future, we might assume it's protected and treat it differently than we might if we knew it used to be public.

Reply:
Fair point. I can add a note instead